examples/gno.land/r/demo/foo721/foo721.gno

```diff
@@ -0,0 +1,122 @@
+package foo721
+
+import (
+	"std"
+
+	"gno.land/p/demo/grc/grc721"
+	"gno.land/p/demo/ufmt"
+	"gno.land/r/demo/users"
+)
+
+var (
+	admin std.Address = "g1us8428u2a5satrlxzagqqa5m6vmuze025anjlj"
+	foo               = grc721.NewBasicNFT("FooNFT", "FNFT")
+)
+
+func init() {
+	mintNNFT(admin, 10)                                     // @administrator (10)
+	mintNNFT("g1var589z07ppjsjd24ukm4uguzwdt0tw7g47cgm", 5) // @hariom (5)
+}
+
+func mintNNFT(owner std.Address, n uint64) {
+	count := foo.TokenCount()
+	for i := count; i < count+n; i++ {
+		tid := grc721.TokenID(ufmt.Sprintf("%d", i))
+		foo.Mint(owner, tid)
+	}
+}
+
+// Getters
+
+func BalanceOf(user users.AddressOrName) uint64 {
+	balance, err := foo.BalanceOf(user.Resolve())
+	if err != nil {
+		panic(err)
+	}
+
+	return balance
+}
+
+func OwnerOf(tid grc721.TokenID) std.Address {
+	owner, err := foo.OwnerOf(tid)
+	if err != nil {
+		panic(err)
+	}
+
+	return owner
+}
+
+func IsApprovedForAll(owner, user users.AddressOrName) bool {
+	return foo.IsApprovedForAll(owner.Resolve(), user.Resolve())
+}
+
+func GetApproved(tid grc721.TokenID) std.Address {
+	addr, err := foo.GetApproved(tid)
+	if err != nil {
+		panic(err)
+	}
+
+	return addr
+}
+
+// Setters
+
+func Approve(user users.AddressOrName, tid grc721.TokenID) {
+	err := foo.Approve(user.Resolve(), tid)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func SetApprovalForAll(user users.AddressOrName, approved bool) {
+	err := foo.SetApprovalForAll(user.Resolve(), approved)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func TransferFrom(from, to users.AddressOrName, tid grc721.TokenID) {
+	err := foo.TransferFrom(from.Resolve(), to.Resolve(), tid)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Admin
```

Comment:
On what is this GRC721 contract being modeled after?

Ethereum NFTs (ERC721 contracts) [don't have these methods](https://wiki.iota.org/shimmer/smart-contracts/guide/evm/examples/ERC721).

I don't see a getter / setter for the NFT URI, which is the whole point of having an NFT

Reply:
I referred https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/ERC721.sol 
And as per my understanding, the metadata is OPTIONAL for ERC-721 smart contracts.